hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -152,7 +152,7 @@ func (p Agent) ReconcileCredentials(ctx context.Context, c client.Client, create
 	roleBinding := &rbacv1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: hcluster.Spec.Platform.Agent.AgentNamespace,
-			Name:      credentialsRBACName,
+			Name:      fmt.Sprintf("%s-%s", credentialsRBACName, controlPlaneNamespace),
```

Comment:
How does this get cleaned up?
/hold

Reply:
actually not as it is in another namespace and possibly we need some finalizer but it is same as now, just allows to have role binding per hosted cluster for the same agent

Reply:
Not sure I follow the reasoning, why would we not cleanup because the thing we create per hostedcluster is in a different namespace?